friends/src/App.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react';
+import { Route } from 'react-router-dom';
+import axios from 'axios';
+import FriendList from './components/FriendList';
+import FriendForm from './components/FriendForm';
+
+class App extends Component {
+  state = {
+    friends: [],
+    id: '',
+    name: '',
+    age: '',
+    email: ''
+  };
+
+  componentDidMount() {
+    this.fetchFriends();
+  }
+
+  fetchFriends() {
+    axios
+      .get('http://localhost:5000/friends')
+      .then(({ data }) => this.setState({ friends: data }))
+      .catch(err => console.error(err));
+  }
+
+  handleFriendSubmit = event => {
+    event.preventDefault();
+
+    const { name, age, email, id } = this.state;
+    const url = id
+      ? `http://localhost:5000/friends/${id}`
+      : 'http://localhost:5000/friends';
+
+    axios[id ? 'put' : 'post'](url, { name, age, email })
```

Comment:
Veeery clever and clean way of handling both! The id will only be present if the friends already exists and therefore the user is attempting to PUT some changes on them. Nice :100: